fabric-player-tracking-api-v1/src/main/java/net/fabricmc/fabric/api/networking/player/tracking/v1/PlayerTracking.java

```diff
@@ -0,0 +1,182 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.networking.player.tracking.v1;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerChunkManager;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.server.world.ThreadedAnvilChunkStorage;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.math.Vec3i;
+import net.minecraft.world.chunk.ChunkManager;
+
+import net.fabricmc.fabric.impl.networking.player.tracking.ThreadedAnvilChunkStorageTrackingExtensions;
+
+/**
+ * For example, a block entity may use the methods in this class to send a packet to all clients which can see the block entity in order notify clients about a change.
+ *
+ * <p>The word "tracking" means that an entity/chunk on the server is known to a player's client (within in view distance) and the (block) entity should notify tracking clients of changes.
+ *
+ * <p>These methods should only be called on the server thread and only be used on logical a server.
+ */
+public final class PlayerTracking {
+	/**
+	 * Gets all the players on the minecraft server.
+	 *
+	 * <p>The returned collection is immutable.
+	 *
+	 * @param server the server
+	 * @return all players on the server
+	 */
+	public static Collection<ServerPlayerEntity> all(MinecraftServer server) {
+		Objects.requireNonNull(server, "The server cannot be null");
+
+		// return an immutable collection to guard against accidental removals.
+		if (server.getPlayerManager() != null) {
+			return Collections.unmodifiableCollection(server.getPlayerManager().getPlayerList());
+		}
+
+		return Collections.emptyList();
+	}
+
+	/**
+	 * Gets all the players in a server world.
+	 *
+	 * <p>The returned collection is immutable.
+	 *
+	 * @param world the server world
+	 * @return the players in the server world
+	 */
+	public static Collection<ServerPlayerEntity> world(ServerWorld world) {
+		Objects.requireNonNull(world, "The world cannot be null");
+		// return an immutable collection to guard against accidental removals.
+		return Collections.unmodifiableCollection(world.getPlayers());
+	}
+
+	/**
+	 * Gets all players tracking a chunk in a server world.
+	 *
+	 * @param world the server world
+	 * @param pos   the chunk in question
+	 * @return the players tracking the chunk
+	 */
+	public static Stream<ServerPlayerEntity> tracking(ServerWorld world, ChunkPos pos) {
```

Comment:
There is some inconsistency between Stream and Collection use. I'd use Collection everywhere since partial consumption is unlikely and it's the simpler API.